controller/src/controller/main_systems/system_monitor.py

```diff
@@ -314,8 +321,32 @@ async def _handle_comm_from_instrument_comm(self) -> None:
                             FW_UPDATE_SUBDIR,
                             f"{firmware_type}-{fw_version}.bin",
                         )
+
                         os.remove(fw_file_path)
+
                     system_state_updates[key] = None
+                case {"command": "init_offline_mode"}:
+                    system_state_updates["system_status"] = SystemStatuses.OFFLINE_STATE
+                case {"command": "end_offline_mode", "stim_info": stim_info}:
+                    # setup stim state entering online mode
+                    system_state_updates["stim_info"] = stim_info
+                    system_state_updates["stimulation_protocol_statuses"] = [StimulationStates.RUNNING] * len(
```

Comment:
Did we say that headless mode can only be entered if all the protocols running are set to run forever? If we're allowing protocols to be run only once and then stop then we'll need to check the stim statuses for each protocol instead of setting all of them to running